Adn/remove occupancies

alexandredamiron opened the pull request:
routeBeginOccupancies are no longer existent and not stored in store. Current bounds are derived from routeAspects

multun committed:
core: make an adapter for the new infra APIs

jacomyal committed:
editor: removes edition zone

Details:
- Removes edition zone selection tool
- Removes zone and entities from editor reducers and state
- Adapts all layer to render the vector data directly from the server
- Adapts all points layers minzoom, to prevent all entities to be
visible from space

TODO:
- Load data from editoast when the new routes become available
- Adapt internal logic for each tool
- Fix zooming speed
- Fix all broken things, basically...

eckter committed:
core: stdcm: fix typo that sometimes caused duplicate routes

eckter committed:
tests: fuzzer: distribute the scheduled trains over a whole day

This adds more variety to the stdcm test cases

jacomyal committed:
editor: removes edition zone

Details:
- Reads actual data from editoast, thanks to new dedicated API route
- EntitySumUp now handles both having an entity as input or just its ID
and type. It also fetches dependencies by itself.
- Improves `hovered` state management

TODO:
- Check and fix dedicated tools

Khoyo committed:
core: implement signaling simulation

Interface changes:
- rename SimInfra into InterlockingInfra
- reference ZonePaths by numerical identifiers

New modules:
- kt-osrd-signaling is a generic signaling simulator
- kt-osrd-sncf-signaling implement SNCF signaling systems and drivers

New interfaces:
- RawSignalingInfra stores the unloaded signaling infrastructure, and inherits from InterlockingInfra
- LoadedSignalInfra stores the loaded signaling infrastructure, which has bound signaling systems and drivers for each signal
- BlockInfra stores the block graph

Co-authored-by: Victor Collod <victor.collod@epita.fr>
Co-authored-by: Younes Khoudli <younes.khoudli@epita.fr>

younesschrifi committed:
editoast: add filters (level, error_type and object_id) to list_errors endpoint

multun committed:
core: fix ArenaMap warning

Khoyo committed:
railjson: add the new signaling model

nicolaswurtz committed:
Nwz/crud infra (#2561)

* front: crudInfra: set infraselector as single source of truth for api

* front: crudInfra: continue isolation of component

* front: crudInfra: added locked state

* front: crudInfra: getInfra inside infraselectormodal

* front: crudInfra: some cosmetics changes, icons, harmonize styles

* front: crudInfra: Added lock/unlock & cosmetics changes

* front: crudInfra: repair side damage on timetable selector

* front: crudInfra: export railjson ok

* front: crudInfra: delete ok

* front: crudInfra: done !

* front: have to desactivate resizing of map in simulation, we'll have to correct modal behaviour before

* front: crudInfra: removed setMustRefresh because it was ugly

alexandredamiron committed:
rebase first study

alexandredamiron committed:
first study

multun committed:
simplify issue templates

These templates have a number of issues:
- the templates assume the only developers working on the project work at SNCF Réseau
- they are deeply unusual. I regularly interact with other bug trackers, and have not seen anything like it in the open source world
- their complexity may scare away contributors
- they don't reflect how SNCF devs actually work. I'm not aware of anyone actually using these templates as intended
- things work just fine, and people only use the description and acceptance criteria part of the templates

eckter committed:
core: remove edge lazy evaluation

turns out this was an early optimization,
it starts to be a problem now

majaziri committed:
Mji/pathfinding optimisation (#2613)

* correct error detection in case of multiple constraints having errors

* fix gradle check errors

clarani committed:
Fix grid margins

alexandredamiron committed:
Front: Adn/add grid margin (#2596)

* prepare isolation

* first Isolation Effort

* Standard Allowance Isolation

* Advanced Stdcm Setup

* set grid Margin After Before in stdcm Context

* prepare isolation

* first Isolation Effort

* Standard Allowance Isolation

* Advanced Stdcm Setup

* set grid Margin After Before in stdcm Context

* Fill with persisted Values

* eslint and exhaustive deps

* Correct optim error

* après trad

* conditionnal layout

sim51 committed:
front: edition - fixing issues (#2614)

* front: editor - adding tooltip on button to select a tracksection on the map #2503

* front: editor - fix issue when clicking on button 'select a tracksection' and you hovera signal for example (bug 4-bis in #2503)

* front: editor - when changing the position of a signal in the form, the item is also moved on the map

* front: editor - fix new entity popin by not displaying default id &
click on line for signals

nicolaswurtz committed:
Nwz/import trains from graou (#2586)

* front: importGraou: first try

* front: importGraou: date & time added

* front: GraouImport: working config, working getTrainsList

* front: GraouImport: trainslist ok, beginning modal import

* front: GraouImport: path complete done

* front: GraouImport: pathfinding ok

* front: GraouImport: adding global settings from osrdconfig

* front: OpenDataImport: switching from Graou to OpenData

* front: OpenDataImport: added rollingstockid to result

* front: OpenDataImport: it's working !

* front: OpenDataImport: using real external serveur from Graou

* front: OpenDataImport: some corrections

* front: OpenDataImport: adapt map to maplibre migration

* front: OpenDataImport: cleaning timetableselector a little

* front: OpenDataImport: direct links to simulation & stdcm after import

* front: OpenDataImport: first pass to make pathfinding by OPs trigram working (and it's working !)

* front: OpenDataImport: some bugfixes

* front: OpenDataImport: some corrections

* front: some bugfixes

* front: Remove DGEX Solutions mentions

* front: fast relooking homepage

* front: rework home page

* front: opendataimport: bugfix

* front: replace png by svg when possible & svgo

* front: OpenDataImport: add defaultProps values

* front: OpenDataImport: bugfix on state selection for mapStyle

* front: OpenDataImport: state selector for timetableID more lightweight

eckter committed:
core: tests: stdcm: add builder for optional parameters

eckter committed:
core: tests: stdcm: split tests into several new files

multun committed:
better issue templates

Fixes #2625

This change does two things:
 - it adds interactive issue templates
 - it moves away from SAFE-specific wording, which is entirely unknown
   to the opensource world: user stories become feature requests, and
   enablers become development tasks

flomonster committed:
codecov: add config

anisometropie committed:
use static initialiser for errors hashmap

flomonster committed:
codeowners: update with new groups

Akctarus committed:
front: dts: fix time display

jacomyal committed:
editor: removes edition zone

Details:
- Adds getMixedEntities to load heterogeneous entities at once
- Fixes selecting with rectangle or polygon, reading rendered features
and then loading them as batches

eckter committed:
api: docker: update base image version

flomonster committed:
editoast: move to rust stable version

multun committed:
core: implement MutableList.clone

Co-authored-by: Younes Khoudli <younes.khoudli@epita.fr>

nicolaswurtz committed:
Nwz/bugfixes and minor cosmetic corrections (#2638)

* front: transform map carto in hook & add new icon

* front: hook & change icon for osrd/timetable

* front: hook & change icon of editor home

* front: hook & change icon for stdcm

* front: change icon for opendataimport

* front: change icon on customget

* front: spelling corrections to please Céline

* front: correct customGet data conversion on stops

* front: CrudInfra: working rename endpoint

* front: spelling corrections

* front: CrudInfra: rework of edition/selection switch

* front: RollingStocks selector: better & smaller UI

* front: RollingStocks: isolate & moving to common

* front: correct checkbox & radio selectors

* front: map: a bit of cleaning

flomonster committed:
root: remove trash file

flomonster committed:
editoast: fix db connections pool size

Commit:
api: schema: add electric profile schema

flomonster committed:
editoast: bump dependencies

- **serde** from 1.0.150 to 1.0.151 (close #2657)
- **serde_json** from 1.0.89 to 1.0.91 (close #2656)
- **thiserror** from 1.0.37 to 1.0.38 (close #2655)
- **enum-map** from 2.4.1 to 2.4.2 (close #2653)

Commit:
api: add electric profile list model

clarani committed:
Return the fastest path

jacomyal committed:
editor: removes edition zone

Details:
- Fixes TrackSection points edition
- Fixes popups not updating when hovering different entities with the
selection tool
- Fixes various cases with the new "getEntity" paradigm

TODO:
- Find why there is no render done on Chrome while moving the mouse
(like a broken debounce or something?)
- Find why various track section layers break MapLibre when using the
tracks edition tool (and restore them)
- Find remaining `entitiesIndex` occurrences and fix them

multun committed:
update the existing code of conduct

The existing code of conduct:
- contains a weird list of unrelated projects in the header
- is not the latest version of the contributor covenant
- is not in the right location, which is why almost nobody noticed

younesschrifi committed:
editoast : update openapi

jacomyal committed:
editor: removes edition zone

Details:
- Fixes duplicate IDs in getEntities
- Fixes SwitchEdition without the selection zone

jacomyal committed:
editor: removes edition zone

Details:
- Remount map when changing tool (very inelegant solution, but it solves
unpredictible MapLibre errors, occurring on tools changes)
- Restore TrackSection disabled layers

jacomyal committed:
editor: removes edition zone

Details:
- Improves typing in editoast API route clients
- Spreads a map reference to the layersComponent component as a prop
- Fixes pointEdition, and adapts it to work without the zone selection

Commit:
api: remove enums for electrical profiles and rename

younesschrifi committed:
editoast :  add created and modified column to an infra

younesschrifi committed:
editoast: add speed limit tags endpoint to editoast

younesschrifi committed:
editoast : add get voltages by infra endpoint

younesschrifi committed:
editoast: add import railjson by id

alexandredamiron committed:
dirty removal

alexandredamiron committed:
dirty removal

anisometropie committed:
api: fix tests database setup

When running tests, django creates a postgres database:
https://docs.djangoproject.com/en/4.1/topics/testing/overview/#the-test-database

To make it work:
- add permissions to create a database to the osrd role
- create the database from the postgis template when running tests

sim51 committed:
front: editor - fixing angle computation on getNearestPoint #2518

sim51 committed:
front: editor - fixing distance computation on signal creation by getting the traksection with the API #2518

sim51 committed:
front: some linter fixes

Commit:
api: add electrical profile to curve conditions

Add migration

flomonster committed:
actions: fix editoast ci (uses wrong cache key)

multun committed:
core: use a library catalog (libs.versions.toml)

Currently, we need to manually sync dependency versions between modules.
Using a centralized dependency catalog enables easier upgrades.
https://docs.gradle.org/current/userguide/platforms.html

bloussou committed:
Remove typo in `core` and `editoast` openapi description (#2747)

This PR is :
- removing useless point and quotes in `core/openapi.yaml`
- fixing typo in `editoast/openapi.yaml`
- using consistent wording between descriptions

anisometropie committed:
front: fix crash (#2748)

Khoyo committed:
core: add gradle-versions-plugin

Khoyo committed:
core: upgrade to the latest stable versions

flomonster committed:
api: bump dependencies

- **pillow** from 9.3.0 to 9.4.0 (close #2738)
- **django** from 4.1.3 to 4.1.5 (close #2737)
- **pydantic** from 1.10.2 to 1.10.4 (close #2730)
- **mock** from 4.0.3 to 5.0.0 (close #2718)
- **numpy** from 1.23.5 to 1.24.1 (close #2713)
- **isort** from 5.10.1 to 5.11.4 (close #2700)
- **sentry-sdk** from 1.11.0 to 1.12.1 (close #2648)
- **geojson-pydantic** from 0.4.3 to 0.5.0 (close #2633)
- **pyproject-flake8** from 5.0.4 to 6.0.0.post1 (close #2632)
- **black** from 22.10.0 to 22.12.0 (close #2555)
- **certifi** from 2022.9.24 to 2022.12.7 (close #2534)
- **django-debug-toolbar** from 3.7.0 to 3.8.1 (close #2494)

flomonster committed:
editoast: bump dependencies

- **serde** from 1.0.151 to 1.0.152 (close #2714)
- **clap** from 4.0.29 to 4.0.32 (close #2711)

multun committed:
api: bump poetry

younesschrifi committed:
editoast: bump infra version when importing infra from railjson

sim51 committed:
front: editor - first implementation of the infra error modal

clarani committed:
api: add meta_data field in rolling stock model

Khoyo committed:
backend: embed version information

flomonster committed:
actions: fix editoast tarpaulin ci step

Castavo committed:
core: railjson: refactor track ranges classes

Some things were a bit odd, some were no longer used.

I needed the name `RJSTrackRange` to be available for an actual track
range (with a track reference in it) so I renamed the former class to
`RJSRange` and propagated the changes.

sim51 committed:
front: adding loader fill

sim51 committed:
front: display infra errors in a module with a table shape

sim51 committed:
front: fixing linter issues on editor

alexandredamiron committed:
Solution for occupancy bounds live
pass trainDetails to ts
Correct routeAspect.

alexandredamiron committed:
Solution for occupancy bounds live
pass trainDetails to ts
Correct routeAspect.

alexandredamiron committed:
remove commented property

dependabot[bot] committed:
editoast: bump tokio from 1.23.0 to 1.24.1 in /editoast

Bumps [tokio](https://github.com/tokio-rs/tokio) from 1.23.0 to 1.24.1.
- [Release notes](https://github.com/tokio-rs/tokio/releases)
- [Commits](https://github.com/tokio-rs/tokio/compare/tokio-1.23.0...tokio-1.24.1)

---
updated-dependencies:
- dependency-name: tokio
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

flomonster committed:
api: editoast: fix health endpoint to openapi

flomonster committed:
editoast: fix endpoint that retrieve infra errors

A diesel error occurred when there was no geometry
of the error.

flomonster committed:
actions: cleanup and improve editoast ci

flomonster committed:
editoast: improve get railjson endpoint code quality

anisometropie commented on front/src/reducers/osrdsimulation/types.ts:
```diff
@@ -52,7 +52,7 @@ interface Stop {
 }
 
-export interface RouteAspect<Time = number, Color = number> {
+export interface RouteAspect<Time = number, Color = string> {
```
By default it’s `RouteAspect<number, number>`. if changed it will affect all the places where color needs to be a number.

if you need to use the other one  (used in `consolidatedSimulation`) it’s a `RouteAspect<Date | null, string>`

I have created type aliases so we don’t have to type those params all the time https://github.com/DGEXSolutions/osrd/pull/2799

anisometropie commented on front/src/applications/osrd/components/Simulation/drawArea.js:
```diff
@@ -2,6 +2,9 @@ import * as d3 from 'd3';
 const drawArea = (chart, classes, dataSimulation, groupID, interpolation, keyValues, rotate) => {
   // Pattern could be a color or a pattern defined in svgDefs with syntax 'url(#idOfPAttern)'
+
+  console.log('draw area', dataSimulation)
```
remove

anisometropie commented on front/src/applications/osrd/views/OSRDSimulation/TrainDetails.tsx:
```diff
@@ -1,14 +1,42 @@
 import { useSelector, useDispatch } from 'react-redux';
 import { useTranslation } from 'react-i18next';
 import { updateStickyBar } from 'reducers/osrdsimulation/actions';
+import { RouteAspect } from '../../../../reducers/osrdsimulation/types';
```
it’s simply 'reducers/osrdsimulation/types'

anisometropie commented on front/src/applications/osrd/views/OSRDSimulation/TrainDetails.tsx:
```diff
@@ -1,14 +1,42 @@
+ * @returns current Canton occupied by current train, [start, end] in meters
+ * @TODO do not work with colors as string as soon as possible
+ */
+const getOccupancyBounds = (routeAspects: RouteAspect[] = [], time = 0) => {
```
use `ConsolidatedRouteAspect`  alias for `RouteAspect<Date | null, string>`

jacomyal committed:
editor: adds missing React key in InfraErrorsList

anisometropie committed:
front: useful type aliases (#2799)

To avoid having to give their params all the time :

```
ConsolidatedMergeDataPoint = MergedDataPoint<Date | null>
ConsolidatedPosition = Position<Date | null>
ConsolidatePositionSpeed = PositionSpeed<Date | null>
ConsolidatedRouteAspect = RouteAspect<Date | null, string>
ConsolidatedSignalAspect = SignalAspect<Date | null, string>
```

alexandredamiron committed:
use alias types

alexandredamiron committed:
Correct conslidated type
